src/github.com/matrix-org/go-neb/services/imgur/imgur.go

```diff
@@ -0,0 +1,270 @@
+// Package imgur implements a Service which adds !commands for imgur image search
+package imgur
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/matrix-org/go-neb/types"
+	"github.com/matrix-org/gomatrix"
+)
+
+// ServiceType of the Imgur service
+const ServiceType = "imgur"
+
+var httpClient = &http.Client{}
+
+type imgurGalleryImage struct {
+	ID          string `json:"id"`          // id	string	The ID for the image
+	Title       string `json:"title"`       // title	string	The title of the image.
+	Description string `json:"description"` // description	string	Description of the image.
+	DateTime    int64  `json:"datetime"`    // datetime	integer	Time inserted into the gallery, epoch time
```

Comment:
Fair enough - I had left them in to match the comments for the unimplemented fields (where it probably makes sense to retain the field name and type, for when we implement them in future).